api/v1alpha3/vspherefailuredomain_types.go

```diff
@@ -0,0 +1,130 @@
+	// name of the Region
+	// if is defined with tags, then should be the tag name, for eg: k8s-region-us-west
+	// if not defined with tags for the case with hostgroup, then it’s a user defined region name
+	Region string `json:"region"`
+
+	// name of the Zone,
+	// if is defined with tags, then should be the tag name, for eg:  k8s-zone-us-west-az1
+	// if mapped to hostgroup, then should be the name of the hostgroup
+	Zone string `json:"zone"`
+
+	// Labels is the Tag category used to configure Region/Zone tags
+	// for FailureDomainDatacenter and FailureDomainComputeCluster
+	// +optional
+	Labels LabelsConfig `json:"labels,omitempty"`
+
+	// The underlying infrastructure for this failure domain
+	// Datacenter as the failure domain
+	Datacenter string `json:"datacenter"`
+
+	// ComputeCluster as the failure domain
+	// +optional
+	ComputeCluster *string `json:"computeCluster,omitempty"`
+
+	// HostGroup as the failure domain
+	// +optional
+	HostGroup *FailureDomainHostGroup `json:"hostGroup,omitempty"`
+
+	// the placement constraints which is used within this failure domain
+	PlacementConstaint PlacementConstraint `json:"placementConstraint"`
+}
+
+// PlacementConstraint is the context information for VM placements within a failure domain
+type PlacementConstraint struct {
+	// ResourcePool is the name or inventory path of the resource pool in which
+	// the virtual machine is created/located.
+	// +optional
+	ResourcePool string `json:"resourcePool,omitempty"`
+
+	// Datastore is the name or inventory path of the datastore in which the
+	// virtual machine is created/located.
+	// +optional
+	Datastore string `json:"datastore,omitempty"`
+
+	// NetworkName is the network name for this machine's VM.
+	NetworkName string `json:"NetworkName"`
+
+	// Folder is the name or inventory path of the folder in which the
+	// virtual machine is created/located.
+	// +optional
+	Folder string `json:"folder,omitempty"`
+}
+
+type LabelsConfig struct {
+	// name of the Region
+	// if is defined with tags, then should be the tag name, for eg: k8s-region-us-west
+	// if not defined with tags for the case with hostgroup, then it’s a user defined region name
+	RegionCategory string `json:"region"`
+
+	// name of the Zone,
+	// if is defined with tags, then should be the tag name, for eg:  k8s-zone-us-west-az1
+	// if mapped to hostgroup, then should be the name of the hostgroup
+	ZoneCategory string `json:"zone"`
+}
+
+// FailureDomainHostGroup as the failure domain
+type FailureDomainHostGroup struct {
+	// name of the host group
+	Name string `json:"name"`
+
+	// compute cluster that this hostgroup belongs to
+	// +optional
+	AutoConfigure *bool `json:"autoConfigure,omitempty"`
+}
+
+// VSphereFailureDomainStatus defines the observed state of VSphereFailureDomain
+type VSphereFailureDomainStatus struct {
+	// Vcenters is the list of vcenters in which
+	// this failure domain is available
+	Vcenters []string `json:"vcenters,omitempty"`
```

Comment:
confused by the vcenters slice. I thought FDs across vcenters was a non-goal from the proposal